src/main/java/by/itech/lab/supplier/controller/UserController.java

```diff
@@ -1,29 +1,83 @@
+import by.itech.lab.supplier.dto.UserDto;
+import by.itech.lab.supplier.service.UserService;
+import by.itech.lab.supplier.service.validator.IdValidator;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
+import org.springframework.http.HttpHeaders;
 import org.springframework.http.HttpStatus;
-import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.CrossOrigin;
+import org.springframework.web.bind.annotation.DeleteMapping;
 import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
 import org.springframework.web.bind.annotation.RequestMapping;
 import org.springframework.web.bind.annotation.ResponseStatus;
 import org.springframework.web.bind.annotation.RestController;
 
+import javax.validation.Valid;
+import java.net.URI;
+import java.net.URISyntaxException;
 import java.util.List;
 
-import static by.itech.lab.supplier.constant.ApiConstants.URL_COMPANIES;
-import static by.itech.lab.supplier.constant.ApiConstants.URL_USER;
-
+@CrossOrigin
 @RestController
-@AllArgsConstructor
-@RequestMapping(URL_USER)
+@RequestMapping(ApiConstants.URL_APPLICATION)
 public class UserController {
+    private final Logger log = LoggerFactory.getLogger(UserController.class);
+
+    private final UserService userService;
+
+    //  private final MailService mailService;
+
+    @Autowired
+    public UserController(UserService userService) {
+        this.userService = userService;
+        //      this.mailService = mailService;
+    }
+
+    @GetMapping("/" + ApiConstants.URL_USER + "/{id}")
+    public ResponseEntity<UserDto> getUser(@PathVariable @Valid Long id) {
+        IdValidator.validate(id);
+        log.debug("request to get User : {}", id);
+        UserDto userDto = userService.getUserWithAuthoritiesById(id).get();
+        return new ResponseEntity<>(userDto, HttpStatus.OK);
+    }
+
+    @GetMapping(ApiConstants.URL_USER)
+    public ResponseEntity<List<UserDto>> getAllUsers(Pageable pageable) {
+        final Page<UserDto> page = userService.getAllManagedUsers(pageable);
+        HttpHeaders headers = PaginationUtil.generatePaginationHttpHeaders(page, "/api/users");
+        return new ResponseEntity<>(page.getContent(), headers, HttpStatus.OK);
+    }
+
+    @PostMapping(ApiConstants.URL_USER)
+    public ResponseEntity<UserDto> createUser(@Valid @RequestBody UserDto userDto) throws URISyntaxException {
+        UserDto newUser = userService.createUser(userDto);
+        // call service to send mail
```

Comment:
mail must be sent in service, as it business logic, in the controller must be only logic for handling request and gathering data